Assistant Overhaul

FlamingLily opened the pull request:
FORUM THREAD: https://forums.aurorastation.org/topic/20580-assistant-overhaul/#comment-179129

This one will probably need a wiki change or two.

The Assistant Overhaul! A.K.A "FlamingLily commits to coding something and forgets for six months"

Also, the "Lab Assistant" (Science's learner role) has been renamed to "Research Intern" with alt-titles to match, in the same vein as the medical intern. This serves the dual role of clarifying the role's nature as a departmental learner, AND makes space for the assistant alt title.

This PR adds four new titles to the Assistant role, complete with access and loadouts to match.
* Wait Staff: This one's pretty self explanatory. They haul food and drinks FROM the kitchen and bar TO customers. They do not cook, and do not get kitchen or bar access.
* Technical Assistant: This one's job is to haul crates and parts for engineers. They get access to engineering's lobby and main hallways (as far as a Janitor can get), but otherwise get nothing. They CANNOT start the engines, shields, atmosphere, or anything. They spawn with a high-vis vest and a debugger, for token vending-machine fixing efforts.
* Lab Assistant: This one's job is to haul crates and equipment for scientists. They get access to ACCESS_RESEARCH which is like, two doors in the research meeting room and not even the hallway outside I think (idk). They also get a labcoat and safety goggles. For safety.
* Medical Orderly: This one's job is to haul equipment and corpses, clean up a messy GTR, and stay out of medical's way when an alert is called. Maybe they can man the reception as well. They spawn with a labcoat and sterilizine (not even cleaner, because it's funnier that way.)

I want to insist that these are assistant alt titles, and not roles with responsibilities. They cannot do repairs, they cannot do science, they cannot do surgery, and they CERTAINLY can't cook.

Changelog:
  - rscadd: "Added four new assistant alt titles: Wait Staff, Technical Assistant, Lab Assistant, and Medical Orderly (These are still ASSISTANTS, and cannot perform the relevant departmental skills. These are also different from learner roles, such as the Interns.)"
  - rscadd: "Renamed the existing "Lab Assistant" (The learner role) to "Research Intern" to better represent the intent for the role and to make space for the alt title."

FlamingLily commented:
!review

FlamingLily commented on code/game/jobs/job/ship_crew.dm:
```diff
@@ -33,6 +53,28 @@
+/obj/outfit/job/assistant/waiter
+	name = "Wait Staff"
+	uniform = /obj/item/clothing/under/waiter
+  
```
```suggestion

```

KingOfThePing commented:
The only suggestion is to swap Lab Assistant and Research Intern around. Interns can be whoever, Lab Assistants usually are undergraduates or something similar. It would make more sense to make the interns the assistant-relevant role.

Sniblet commented:
A revolution in viax and failson gameplay alike! If someone shoots this down, I will make a threat and act on that threat.

FlamingLily commented:
> The only suggestion is to swap Lab Assistant and Research Intern around. Interns can be whoever, Lab Assistants usually are undergraduates or something similar. It would make more sense to make the interns the assistant-relevant role.

I'm not sure that's a great idea, personally. The reason I changed [SCIENCE LEARNER] from "Lab Assistant" to "Research Intern" was to bring it in line with the Medical Intern and make it more clear that they're the actual limited learner roles. [SCIENCE LEARNER] as a role is the only learner role who's purpose is sort of unclear. Mainly, the occupational requirements don't require the character to be in education, like the other departmental learners (and this is reflected mechanically as the char age requirement for it is 18)

Under the current setup, it's entirely justified to make a character who's designed to never transition from a learner to an actual departmental role, and this is undesirable (as it takes a slot away from actual learners), which is the main reason I split it up to the learner role and the permanent-assistant title.

However, if the rename of Lab Assistant is a bit too confusing, I could also make the assistant alt title "Lab Technician" while keeping the learner role "Research Intern"

NonQueueingMatt reviewed (approved):
I guess it's worth a go even if I'm a little suspicious of how this can turn into stepping over boundaries pretty quickly (particularly for wait staff).
